lib/ui/home/home_screen.dart

```diff
@@ -29,29 +60,52 @@ class _HomeScreenState extends ConsumerState<HomeScreen> {
   Widget build(BuildContext context) {
     ref.listen<HomeViewState>(homeViewModelProvider, (_, state) {
       state.maybeWhen(
-          getUserProfileSuccess: (profile) => {
-                ref.read(_profileAvatarProvider.notifier).state =
-                    profile.avatarUrl,
-              },
-          loading: () {
-            ref.read(_loadingStateProvider.notifier).state = true;
-          },
-          error: (message) => {
-                ref.read(_loadingStateProvider.notifier).state = false,
-                showToastMessage(message)
-              },
+          success: () => {},
+          loading: () {},
+          error: (message) => {showToastMessage(message)},
           orElse: () => {});
     });
+    bool isLoading = ref.watch(profileStream).value == null;
     return Scaffold(
-      body: SafeArea(
-        child: Consumer(
-          builder: (context, widgetRef, _) {
-            return HomeHeader(
-              date: DateTime.now().getFormattedString(),
-              avatar: widgetRef.watch(_profileAvatarProvider),
-            );
+        body: isLoading
+            ? const SurveyShimmerLoading()
+            : _buildHomeContent(mockSurveys));
+  }
+
+  Widget _buildHomeContent(List<SurveyModel> surveys) {
+    final profile = ref.watch(profileStream).value;
+    final today = ref.watch(todayStream).value;
+    return Stack(
+      children: [
+        SurveyView(
+          surveys: surveys,
+          onPageChanged: (visibleIndex) {
+            ref
+                .read(homeViewModelProvider.notifier)
+                .setVisibleSurveyIndex(visibleIndex);
           },
+          onSurveySelected: (survey) {
+            // TODO: Navigate to survey details
+          },
+        ),
+        SafeArea(
+          child: HomeHeader(
+            date: today ?? '',
+            avatar: profile?.avatarUrl ?? '',
+          ),
         ),
+        _pagerIndicator(mockSurveys.length),
+      ],
+    );
```

Comment:
```suggestion
    return Consumer(
      builder: ((context, widgetRef, child) {
        final profile = widgetRef.watch(profileStream).value;
        final today = widgetRef.watch(todayStream).value;
        return Stack(
          children: [
            SurveyView(
              surveys: surveys,
              onPageChanged: (visibleIndex) {
                widgetRef
                    .read(homeViewModelProvider.notifier)
                    .setVisibleSurveyIndex(visibleIndex);
              },
              onSurveySelected: (survey) {
                // TODO: Navigate to survey details
              },
            ),
            SafeArea(
              child: HomeHeader(
                date: today ?? '',
                avatar: profile?.avatarUrl ?? '',
              ),
            ),
            _pagerIndicator(mockSurveys.length),
          ],
        );
      }),
    );
```
It's good to create habits to limit the scope of rebuilding widget here 😄 By using `ref` from HomeScreen, the whole screen will be rebuilt every time a value in `ref.watch` is updated.

Reply:
@manh-t Oh, nice 👍